apps/console/src/features/roles/components/edit-role/edit-role-groups.tsx

```diff
@@ -71,233 +59,166 @@ export const RoleGroupsList: FunctionComponent<RoleGroupsPropsInterface> = (
-        }
+    const {
+        data: applicationData,
+        isLoading: isApplicationRequestLoading,
+        error: applicationRequestError
+    } = useGetApplication(role.audience?.value, roleAudience === RoleAudienceTypes.APPLICATION);
 
-        // Set groups options when the user is read only
-        if (isReadOnly && initialSelectedGroupsOptions) {
-            setGroupsOptions(initialSelectedGroupsOptions);
-        }
-    }, [ isReadOnly, initialSelectedGroupsOptions ]);
+    const excludedIDPs: string[] = [
+        AuthenticatorManagementConstants.ORGANIZATION_ENTERPRISE_AUTHENTICATOR_ID,
+        AuthenticatorManagementConstants.LEGACY_EMAIL_OTP_AUTHENTICATOR_ID,
+        AuthenticatorManagementConstants.LEGACY_SMS_OTP_AUTHENTICATOR_ID
+    ];
 
     /**
-     * Set groups options
+     * Filter out the IDPs.
      */
     useEffect(() => {
-        if (!isReadOnly) {
-            if (originalGroupList && originalGroupList?.totalResults !== 0) {
-                setGroupsOptions(originalGroupList?.Resources);
-            } else {
-                setGroupsOptions([]);
-            }
+        if (isIdPListRequestLoading || idpListError) {
+            return;
         }
-    }, [ originalGroupList ]);
 
-    /**
-     * Set removed groups
-     */
-    useEffect(() => {
-        if (!isReadOnly && initialSelectedGroupsOptions && selectedGroupsOptions) {
-            setRemovedGroupsOptions(initialSelectedGroupsOptions?.filter((group: GroupsInterface) => {
-                return selectedGroupsOptions?.find(
-                    (selectedGroup: GroupsInterface) => selectedGroup.id === group.id) === undefined;
-            }));
+        if (roleAudience === RoleAudienceTypes.APPLICATION
+            && (isApplicationRequestLoading || applicationRequestError)) {
+            return;
         }
-    }, [ initialSelectedGroupsOptions, selectedGroupsOptions ]);
+    
+        const filteredList: StrictIdentityProviderInterface[] = idpList.identityProviders.filter(
+            (provider: StrictIdentityProviderInterface) => {
+                return !excludedIDPs.includes(provider.federatedAuthenticators.defaultAuthenticatorId);
+            }
+        );
+        
+        // If the role is an application role, filter out the IDPs that are configured for the application.
+        if (roleAudience === RoleAudienceTypes.APPLICATION) {
+            const applicationIDPList: StrictIdentityProviderInterface[] = [];
 
-    /**
-     * Call mutateGroupListFetchRequest when the group search value changes
-     */
-    useEffect(() => {
-        if ( groupSearchValue ) {
-            mutateGroupListFetchRequest().finally(() => {
-                setGroupSearchLoading(false);
+            applicationData.authenticationSequence.steps.forEach((step: AuthenticationStepInterface) => {
+                step.options.forEach((option: AuthenticatorInterface) => {
+                    const idp: StrictIdentityProviderInterface = filteredList.find(
```

Comment:
```suggestion
                    const idp: StrictIdentityProviderInterface = filteredList?.find(
```

Reply:
Null check is not needed here. `filteredList` won't be undefined or null